src/chai/ArrayManager.cpp

```diff
@@ -359,14 +358,28 @@ PointerRecord* ArrayManager::makeManaged(void* pointer,
                                          ExecutionSpace space,
                                          bool owned)
 {
+  if (space == NONE) {
+    space = getDefaultAllocationSpace();
+  }
+
   m_resource_manager.registerAllocation(
       pointer,
       {pointer, size, m_allocators[space]->getAllocationStrategy()});
 
   auto pointer_record = getPointerRecord(pointer);
+
   if (pointer_record == &s_null_record) {
-     pointer_record = new PointerRecord();
+     if (pointer) {
```

Comment:
Should we also check for size > 0, too?

Reply:
I don't think so. If we have a non null pointer we do not want to be returning the null pointer record regardless of size.

Reply:
OK.